x/mint/keeper/keeper_test.go

```diff
@@ -60,109 +60,145 @@ func TestKeeperTestSuite(t *testing.T) {
-		feePool.CommunityPool.AmountOf("stake"))
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.DeveloperRewards).TruncateInt(),
-		suite.App.BankKeeper.GetBalance(suite.Ctx, devRewardsReceiver, "stake").Amount)
+			coin := suite.App.MintKeeper.GetProportions(suite.Ctx, tc.fee, tc.ratio)
+			suite.Equal(tc.expectedCoin, coin)
+		})
+	}
+}
 
-	// Test for multiple dev reward addresses
-	params.WeightedDeveloperRewardsReceivers = []types.WeightedAddress{
+func (suite *KeeperTestSuite) TestDistrAssetToDeveloperRewardsAddr() {
+	var (
+		distrTo = lockuptypes.QueryCondition{
+			LockQueryType: lockuptypes.ByDuration,
+			Denom:         "lptoken",
+			Duration:      time.Second,
+		}
+		params       = suite.App.MintKeeper.GetParams(suite.Ctx)
+		gaugeCoins   = sdk.Coins{sdk.NewInt64Coin("stake", 10000)}
+		gaugeCreator = sdk.AccAddress([]byte("addr2---------------"))
+		mintLPtokens = sdk.Coins{sdk.NewInt64Coin(distrTo.Denom, 200)}
+	)
+
+	tests := []struct {
+		name              string
+		weightedAddresses []types.WeightedAddress
+		mintCoin          sdk.Coin
+	}{
+		{
+			name: "one dev reward address",
+			weightedAddresses: []types.WeightedAddress{
+				{
+					Address: sdk.AccAddress([]byte("addr1---------------")).String(),
+					Weight:  sdk.NewDec(1),
+				}},
+			mintCoin: sdk.NewCoin("stake", sdk.NewInt(10000)),
+		},
 		{
-			Address: devRewardsReceiver2.String(),
-			Weight:  sdk.NewDecWithPrec(6, 1),
+			name: "multiple dev reward addresses",
+			weightedAddresses: []types.WeightedAddress{
+				{
+					Address: sdk.AccAddress([]byte("addr3---------------")).String(),
+					Weight:  sdk.NewDecWithPrec(6, 1),
+				},
+				{
+					Address: sdk.AccAddress([]byte("addr4---------------")).String(),
+					Weight:  sdk.NewDecWithPrec(4, 1),
+				}},
+			mintCoin: sdk.NewCoin("stake", sdk.NewInt(100000)),
 		},
 		{
-			Address: devRewardsReceiver3.String(),
-			Weight:  sdk.NewDecWithPrec(4, 1),
+			name:              "nil dev reward address",
+			weightedAddresses: nil,
+			mintCoin:          sdk.NewCoin("stake", sdk.NewInt(100000)),
 		},
 	}
-	suite.App.MintKeeper.SetParams(suite.Ctx, params)
-
-	err = mintKeeper.MintCoins(suite.Ctx, mintCoins)
-	suite.NoError(err)
-	err = mintKeeper.DistributeMintedCoin(suite.Ctx, mintCoin)
-	suite.NoError(err)
+	for _, tc := range tests {
+		suite.Run(tc.name, func() {
+			suite.Setup()
+			mintKeeper := suite.App.MintKeeper
+			// set WeightedDeveloperRewardsReceivers
+			params.WeightedDeveloperRewardsReceivers = tc.weightedAddresses
+			mintKeeper.SetParams(suite.Ctx, params)
+
+			// mints coins so supply exists on chain
+			suite.FundAcc(gaugeCreator, gaugeCoins)
+			suite.FundAcc(gaugeCreator, mintLPtokens)
+
+			gaugeId, err := suite.App.IncentivesKeeper.CreateGauge(suite.Ctx, true, gaugeCreator, gaugeCoins, distrTo, time.Now(), 1)
+			suite.NoError(err)
+			err = suite.App.PoolIncentivesKeeper.UpdateDistrRecords(suite.Ctx, poolincentivestypes.DistrRecord{
```

Comment:
Hi @vuong177 . Awesome work on this! One last step that I would like to ask before merging is if we could change all the keepers to follow this pattern:
```
intencentuvesKeeper := suite.App.IncentivesKeeper
....
err = incentivesKeeper.UpdateDistrRecords(...)
```

I think we should do this for:
- `suite.App.IncentivesKeeper`
- `suite.App.PoolIncentivesKeeper`
- `suite.App.BankKeeper`

We are trying to maintain the new standards with the aim to reduce verbosity stemming from dereferencing keepers from the app multiple times. 

Once this is addressed, should be good to go :+1: Thanks for the hard work!